plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt

```diff
@@ -11,9 +11,11 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhisp
 import software.aws.toolkits.jetbrains.services.codewhisperer.util.CodeWhispererConstants
 
 internal class CodeWhispererCodeScanFileListener(val project: Project) : EditorFactoryListener {
+    val scanManager = CodeWhispererCodeScanManager.getInstance(project)
     override fun editorCreated(event: EditorFactoryEvent) {
+        scanManager.addListeners()
```

Comment:
fix infinite cycle exception by retrieving service instance inside of `editorCreated`